common/types.go

```diff
@@ -58,6 +61,15 @@ func (h *Hash) UnmarshalJSON(input []byte) error {
 	if length >= 2 && input[0] == '"' && input[length-1] == '"' {
 		input = input[1 : length-1]
 	}
+	// strip "0x" for length check
+	if len(input) > 1 && string(input[:2]) == "0x" {
+		input = input[2:]
+	}
+
+	// validate the length of the input hash
+	if len(input) != HashLength*2 {
+		return hashJsonLengthErr
+	}
 	h.SetBytes(FromHex(string(input)))
```

Comment:
I'll do that in my PR. Don't add capital X just here
On Apr 1, 2016 1:43 PM, "Jeffrey Wilcke" <notifications@github.com> wrote:

> In common/types.go
> <https://github.com/ethereum/go-ethereum/pull/2404#discussion_r58189530>:
>
> > @@ -58,6 +61,15 @@ func (h *Hash) UnmarshalJSON(input []byte) error {
> >  	if length >= 2 && input[0] == '"' && input[length-1] == '"' {
> >  		input = input[1 : length-1]
> >  	}
> > +	// strip "0x" for length check
> > +	if len(input) > 1 && string(input[:2]) == "0x" {
> > +		input = input[2:]
> > +	}
> > +
> > +	// validate the length of the input hash
> > +	if len(input) != HashLength*2 {
> > +		return hashJsonLengthErr
> > +	}
> >  	h.SetBytes(FromHex(string(input)))
>
> Oh, there's another issue actually: 0x != 0X since we should support both
> versions.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/ethereum/go-ethereum/pull/2404/files/cdd98e7c08359041ea8ea58f5a8e1d52c6121f71#r58189530>
>